keda/values.yaml

```diff
@@ -153,17 +153,19 @@ securityContext:
       - ALL
     allowPrivilegeEscalation: false
     readOnlyRootFilesystem: true
-    seccompProfile:
-      type: RuntimeDefault
+    # k8s >= 1.19
```

Comment:
Let's not punish those using Kubernetes 1.20 or above; we should use the .Capabilties functionality to see if seccompProfile is supported.

https://github.com/Azure/api-management-self-hosted-gateway/blob/94217377b23c9c0571b2dc90a1dfcced3ca8aa73/helm-charts/azure-api-management-gateway/templates/ingress-stable.yaml#L1

Reply:
Can I get `seccompProfile` support from capabilities? I thought it gives only API support but not at this level

Reply:
I guess that I can check k8s version and ignore the value in case of not supported, but that's worse for me because I'd be changing given user fields. Honestly, I prefer to comment the field (in v2.8 only) but I can't add a note saying that the value will be ignored in k8s <= 1.18 and continue with the previous default value

Reply:
Why not check if Kubernetes version is v1.19 or above and only then apply seccompProfile? That is more user friendly than assume people will see the note and manually apply it

Reply:
Because people can expect it applied in clusters <1.19 as they set the values and we would magically remove it.
Also this change won't go to main, it's only for v2.8 release (which in theory should have a short life)
But I can do your proposal if you prefer

Reply:
> Because people can expect it applied in clusters <1.19 as they set the values and we would magically remove it.

That is fine; it is skipping a configuration because it is not supported on their cluster which is safe to do. If we want to be transparent; we can add the same if in NOTES.txt and mention it was skipped

> Also this change won't go to main, it's only for v2.8 release (which in theory should have a short life)

That's fine; but we still need to do proper fixes.

> But I can do your proposal if you prefer

I think this should be done, but let's ask @zroubalik for a 3rd opinion.